src/core/App.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @author Lewy Blue / https://github.com/looeee
+ *
+ * parameters = {
+ *
+ *  canvas: <canvas> DOM element,
+ *
+ *  scene: new THREE.Scene(),
+ *
+ *  camera: new THREE.Camera( ... )
+ *
+ *  renderer: new THREE.WebGLRenderer( ... )
+ *
+ * }
+ */
+
+function App( parameters ) {
+
+	parameters = parameters || {};
+
+	if ( parameters.canvas !== undefined ) {
+
+	  this.canvas = parameters.canvas;
+
+	}	else {
+
+		this.canvas = document.body.appendChild( document.createElement( 'canvas' ) );
+		this.canvas.style.width = this.canvas.style.height = '100%';
+
+	}
+
+	this.scene = ( parameters.scene !== undefined ) ? parameters.scene : new THREE.Scene();
+
+	this.camera = ( parameters.camera !== undefined ) ? parameters.camera : new THREE.PerspectiveCamera( 75, this.canvas.clientWidth / this.canvas.clientHeight, 1, 1000 );
+
+	if ( parameters.renderer !== undefined ) {
+
+	  this.renderer = parameters.renderer;
+
+	}	else {
+
+	  this.renderer = new THREE.WebGLRenderer( { canvas: this.canvas } );
+	  this.renderer.setSize( this.canvas.clientWidth, this.canvas.clientHeight );
+
+	}
+
+}
+
+Object.assign( App.prototype, {
+
+	animate: function () {
+
+	  var self = this;
+
+	  function animationHandler() {
+
+	    self.currentAnimationFrameID = requestAnimationFrame( function () { animationHandler() } );
```

Comment:
IMO `requestAnimationFrame` should always be the last thing in the loop not the first. As @mrdoob MrDoob pointed out I'm the idiot that suggested it should be at the top in the first place. I was wrong. With it at the top it's often very hard to debug because even if there's an error things just keep going since a new animation frame has already been requested. With it at the bottom your program correctly stops so you can fix things.